src/numbers/signed_integer/i256.cairo

```diff
@@ -164,9 +164,9 @@ fn i256_check_sign_zero(x: i256) {
 /// Cf: IntegerTrait::new docstring
 fn i256_new(mag: u256, sign: bool) -> i256 {
     if sign == true {
-        assert(mag <= BoundedInt::max() / 2, 'int: out of range');
+        assert(mag <= (BoundedInt::max() - 1) / 2, 'int: out of range');
```

Comment:
Thank you for your work, Musa! 
How do you justify this change and why do you subtract 1?

The `BoundInt::Max() `gives the highest allowed number.
Let's look at the example with `u8`. The BoundInt::Max() for `u8` is 255, which is `(2**8) - 1`.
Therefore, the `BoundInt::Max()` for `u256` is `(2**256) - 1`.

If the sign is `true` (negative number), our maximum is half of the maximum of a `u256`.
Now, if the sign is `false` (positive number), our maximum is also half of the maximum of a `u256`, but we subtract one because we consider 0

Reply:
Thank you for the review

`BoundInt::max()` will always give an odd number, I subtracted `1` before dividing by `2` to avoid irregularity.

Reply:
```suggestion
        assert(mag <= 57896044618658097711785492504343953926634992332820282019728792003956564819968_u256, 'int: out of range');
```
Let's just use the value of 2**255 to avoid any confusions as we do with the other signed integers. And let's keep the fix of the tests you have already done.